File: android-scala-maven-single/test/src/main/scala/TempActivityIT.scala
package com.mycompanyname.bananas

import android.test.ActivityInstrumentationTestCase2
import android.view.View
import android.widget.Button
import com.jayway.android.robotium.solo.Solo

class TempActivityIT extends ActivityInstrumentationTestCase2[TempActivity]("com.mycompanyname.bananas", classOf[TempActivity]) {

  lazy private val solo = new Solo(getInstrumentation(), getActivity())

  override protected def tearDown() {
    solo.finishOpenedActivities
    super.tearDown
  }

  @inline def find[T <: View](resId: Int): T = solo.getCurrentActivity.findViewById(resId).asInstanceOf[T]
  lazy private val button = find[Button](R.id.btn)

  def testStuff {
    solo.clickOnView(button);
  }
}

File: android-java-maven-single/test/src/com/mycompanyname/bananas/TempActivityIT.java
package com.mycompanyname.bananas;

import android.test.ActivityInstrumentationTestCase2;
import android.view.View;
import android.widget.Button;
import com.jayway.android.robotium.solo.Solo;
import lombok.val;

public class TempActivityIT extends ActivityInstrumentationTestCase2<TempActivity_> {

	private Solo solo;

	public TempActivityIT() {
		super("com.mycompanyname.bananas", TempActivity_.class);
	}

	@Override
	public void setUp() throws Exception {
		super.setUp();
		solo = new Solo(getInstrumentation(), getActivity());
	}

	@Override
	protected void tearDown() throws Exception {
		solo.finishOpenedActivities();
		super.tearDown();
	}

	public void test() throws Exception {
		val button = (Button) solo.getCurrentActivity().findViewById(R.id.btn);
		solo.clickOnView(button);
	}
}

